src/rgw/rgw_json_enc.cc

```diff
@@ -555,13 +559,22 @@ void rgw_bucket::dump(Formatter *f) const
 }
 
 void rgw_bucket::decode_json(JSONObj *obj) {
-  JSONDecoder::decode_json("name", name, obj);
+  string bucket_name;
+  JSONDecoder::decode_json("name", bucket_name, obj);
   JSONDecoder::decode_json("pool", data_pool, obj);
   JSONDecoder::decode_json("data_extra_pool", data_extra_pool, obj);
   JSONDecoder::decode_json("index_pool", index_pool, obj);
   JSONDecoder::decode_json("marker", marker, obj);
   JSONDecoder::decode_json("bucket_id", bucket_id, obj);
   JSONDecoder::decode_json("tenant", tenant, obj);
+
+  auto pos = bucket_name.find(":");
+  if (pos != string::npos) {
+    name = std::move(bucket_name.substr(0, pos));
+    ns = std::move(bucket_name.substr(pos + 1));
+  } else {
+    name = std::move(bucket_name);
+  }
```

Comment:
consider making this into a helper function, i.e. `set_pool_ns()`, so we don't duplicate this logic in `decode_json()`